### PR TITLE
Metaweather api call on iOS

### PR DESCRIPTION
**Pull request by RamziJabali**
### 💬 Description
iOS now shows loading view, then shows a weather report, also has error handling.

### 🌉 Screenshot

iOS_app_running

### 👷‍♂️ Testing
- `./gradlew :app:build`
- `./gradlew :shared:build`
- `xcodebuild -workspace ios/KaMPKitiOS.xcworkspace -scheme KaMPKitiOS 
    -sdk iphoneos -configuration Debug build -destination name="iPhone 8"`

### CODE REVIEW

File: ios/KaMPKitiOS/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,96 @@
+//
+//  ContentView.swift
+//  KaMPKitiOS
+//
+//  Created by Ramzi Jabali on 1/12/22.
+//  Copyright © 2022 Touchlab. All rights reserved.
+//
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:7]: remove comments

File: ios/KaMPKitiOS/ContentView.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+import SwiftUI
+import shared
+
+private let log = koin.loggerWithTag(tag: "ViewController")
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:12]: ViewController -> ObservableWeatherModel

File: ios/KaMPKitiOS/ContentView.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+         }.position(x: 220, y: 200)
+     }
+}
+
+struct WeatherReportView: View {
+    var weatherReport: WeatherReport?
+
+    var body: some View {
+        VStack {
+            Text(weatherReport?.cityTitle ?? "Loading").bold().font(Font.custom("", size: 60.0))
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:83]: Font.custom("", size: 60.0) -> move to design tokens!

File: ios/KaMPKitiOS/ContentView.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            Text(weatherReport?.countryTitle ?? "Loading").bold()
+            HStack {
+                Text(weatherReport?.temperature ?? "Loading").padding(30)
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:86]: padding value(30) -> move to design tokens!

File: ios/KaMPKitiOS/ContentView.swift [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+     }
+}
+
+struct WeatherReportView: View {
+    var weatherReport: WeatherReport?
+
+    var body: some View {
+        VStack {
+            Text(weatherReport?.cityTitle ?? "Loading").bold().font(Font.custom("", size: 60.0))
+            Text(weatherReport?.countryTitle ?? "Loading").bold()
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:84]: Specify font from design tokens

File: ios/KaMPKitiOS/ContentView.swift [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+struct WeatherViewContent: View {
+    var loading: Bool
+    var weatherReport: WeatherReport?
+    var error: String?
+    
+    var body: some View {
+         HStack {
+             WeatherReportView(weatherReport: weatherReport)
+         }.position(x: 220, y: 200)
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:74]: Replace hardcoded positioning to relative centering

File: ios/KaMPKitiOS/ContentView.swift [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+
+struct WeatherViewContent: View {
+    var loading: Bool
+    var weatherReport: WeatherReport?
+    var error: String?
+    
+    var body: some View {
+         HStack {
+             WeatherReportView(weatherReport: weatherReport)
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:73]: inline WeatherReportView

File: ios/KaMPKitiOS/ContentView.swift (hunk 3 above)
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:83]: "Loading" -> String Resources

File: ios/KaMPKitiOS/ContentView.swift [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        VStack {
+            Text(weatherReport?.cityTitle ?? "Loading").bold().font(Font.custom("", size: 60.0))
+            Text(weatherReport?.countryTitle ?? "Loading").bold()
+            HStack {
+                Text(weatherReport?.temperature ?? "Loading").padding(30)
+                Text(weatherReport?.humidity ?? "Loading").padding(30)
+            }
+            HStack {
+                Text(weatherReport?.windSpeed ?? "Loading").padding(30)
+                Text(weatherReport?.airPressure ?? "Loading").padding(30)
```
Comment (RamziJabali) [inline ios/KaMPKitiOS/ContentView.swift:91]: Loading only for cityTitle

File: shared/src/iosMain/kotlin/co/touchlab/kampkit/ViewModel.kt [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class ViewModel(
+    private val onDataState: (DataState<WeatherReport>) -> Unit
+) : KoinComponent {
+    private val log: Logger by injectLogger("WeatherUseCase")
```
Comment (RamziJabali) [inline shared/src/iosMain/kotlin/co/touchlab/kampkit/ViewModel.kt:20]: "WeatherUseCase" -> LandingPageViewModel

File: shared/src/iosMain/kotlin/co/touchlab/kampkit/ViewModel.kt [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.launch
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class ViewModel(
```
Comment (RamziJabali) [inline shared/src/iosMain/kotlin/co/touchlab/kampkit/ViewModel.kt:17]: ViewModel -> LandingPageViewModel

**Review by seljabali — APPROVED** [review]
Looks good. Good job!

**Review by RamziJabali — COMMENTED** [review]
Completed!

**Pull request merged**